pkg/controller/machinedeployment/controller.go

```diff
@@ -368,6 +371,12 @@ func (c *MachineDeploymentControllerImpl) enqueue(d *v1alpha1.MachineDeployment)
 		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %#v: %v", d, err))
 		return
 	}
+	if _, queueExists := c.informers.WorkerQueues["MachineDeployment"]; !queueExists {
+		glog.V(2).Infof("MachineDeployment queue does not exist, requing after %v", requeueAfterWhenQueueAbsent)
+		time.Sleep(requeueAfterWhenQueueAbsent)
+		c.enqueue(d)
```

Comment:
Shouldn't here have an error out after multiple attempts or timeout?

Reply:
The only case where this happens is during startup of the controller because init is called before the workqeue is created:

https://github.com/kubernetes-sigs/cluster-api/blob/5b128c0f9ab134087d1331be8176099201055a36/pkg/controller/machinedeployment/zz_generated.api.register.go#L59-L71

The init of this controller however registers already eventHandlers for machineSets which will subsequently try to put their owning machineDeployments in the not yet existing qeue: https://github.com/kubernetes-sigs/cluster-api/blob/5b128c0f9ab134087d1331be8176099201055a36/pkg/controller/machinedeployment/controller.go#L65-L69

Just creating the qeue before calling Init is unfortunatelly not possible because all of that happens in generated code.

Since this should only happen during the startup in just a very small timeframe I think its okay like this. WDYT?

Reply:
@pwittrock - is this something that gets fixed with the kubebuilder change?

Reply:
@roberthbailey Are you fine with merging this to have the `machineDeployment` controller working again?

Reply:
Yes, sorry for the delay.